doc/users/installing.rst

```diff
@@ -0,0 +1,333 @@
+`ffmpeg <https://www.ffmpeg.org/>`__/`avconv <https://libav.org/avconv.html>`__ or `mencoder <http://www.mplayerhq.hu/design7/news.html>`__
+    Required for the animation module to be save out put to movie
+    formats.
+
+`ImageMagick <http://www.imagemagick.org/>`__
+    Required for the animation module to be able to save to animated gif.
+
+Optional dependencies
+^^^^^^^^^^^^^^^^^^^^^
+
+`Pillow <http://python-pillow.github.io/>`__
+    If Pillow is installed, matplotlib can read and write a larger
+    selection of image file formats.
+
+`pkg-config <http://www.freedesktop.org/wiki/Software/pkg-config/>`__
+    A tool used to find required non-python libraries.  This is not strictly
+    required, but can make installation go more smoothly if the libraries and
+    headers are not in the expected locations.
+
+
+Required libraries that ship with matplotlib
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+:term:`agg` 2.4
+    The antigrain C++ rendering engine.  matplotlib links against the
+    agg template source statically, so it will not affect anything on
+    your system outside of matplotlib.
+
+`qhull` 2012.1
+    A library for computing Delaunay triangulations.
+
+`ttconv`
+   truetype font utility
+
+six 1.9.0
+    Python 2/3 compatibility library.  Do not use this in third-party
+    code.
+
+
+.. _build_linux:
+
+Building on Linux
+-----------------
+
+It is easiest to use your system package manager to install the dependencies.
+
+If you are on Debian/Ubuntu, you can get all the dependencies
+required to build matplotlib with::
+
+   sudo apt-get build-dep python-matplotlib
+
+If you are on Fedora/RedHat, you can get all the dependencies required
+to build matplotlib by first installing ``yum-builddep`` and then
+running::
+
+   su -c "yum-builddep python-matplotlib"
+
+This does not build matplotlib, but it does get and install the
+build dependencies, which will make building from source easier.
+
+
+.. _build_osx:
+
+Building on OSX
+---------------
+
+The build situation on OSX is complicated by the various places one
+can get the libpng and freetype requirements (darwinports, fink,
+/usr/X11R6) and the different architectures (e.g., x86, ppc, universal) and
+the different OSX version (e.g., 10.4 and 10.5). We recommend that you build
+the way we do for the OSX release: get the source from the tarball or the
+git repository and follow the instruction in :file:`README.osx`.
+
+
+.. _build_windows:
+
+Building on Windows
+-------------------
+
+The Python shipped from http://www.python.org is compiled with Visual Studio
+2008 for versions before 3.3, Visual Studio 2010 for 3.3 and 3.4, and
+Visual Studio 2015 for 3.5.  Python extensions are recommended to be compiled
+with the same compiler.
+
+Since there is no canonical Windows package manager, the methods for building
+freetype, zlib, and libpng from source code are documented as a build script
+at `matplotlib-winbuild <https://github.com/jbmohler/matplotlib-winbuild>`_.
```

Comment:
There is also a `build_alllocal.cmd` which has some comments on adding a conda env and then builds it in such an env.